lib/pbench/test/unit/server/query_apis/test_datasets_list.py

```diff
@@ -196,9 +190,32 @@ def test_http_error(
         }
         index = self.build_index(server_config, ("2020-08",))
         query_api(
+            "/datasets/list",
+            "/_search?ignore_unavailable=true",
             json,
             index,
             exceptions["status"],
-            server_config,
-            exc=exceptions["exception"],
+            body=exceptions["exception"],
+        )
+
+    @pytest.mark.parametrize("errors", (400, 500, 409))
```

Comment:
While it's true (at the moment) that, for each of these errors, the Pbench server response is the same (`BAD_GATEWAY`), this seems likely (to me...) to change...so, wouldn't it be good to have each parameter be a pair with the ES response status and the expected Pbench response status?

(Ditto for the other test(s) with this structure.)

Reply:
Yes, I briefly considered that; however _if_ we change from `BAD_GATEWAY` it's most likely still going to handle all unexpected Elasticsearch 400/500 class errors (`HTTPError`) the same, so such generality would be wasted. If it does become necessary, I'd prefer to defer that effort to the point where we know what we actually need.